app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+  def index
+    @categories = Category.order(name: :asc)
+    @merchants = User.joins(:products).uniq.sort
+
+    scope = Product.where(retired: false).order(name: :asc)
+
+    if params[:category].present?
+      scope = scope.joins(:categories).where(categories: {id: params[:category]})
+    end
+
+    if params[:merchant].present?
+      scope = scope.where(user_id: params[:merchant])
+    end
+
+    @products = scope
+  end
+
+  def show
+    @product = Product.find(params[:id])
+    @order_item = OrderItem.new(product_id: @product.id)
+    @reviews = Review.where(product_id: @product.id)
+    @categories = @product.categories
+  end
+
+  def  search
+    product_names = Product.all.pluck(:name)
+    result = product_names.select do |name|
+      name.include? params[:q].capitalize
```

Comment:
Params is a bucket with all sorts of things in it, you want everything in there to be clearly named. I believe from what I've seen that this might be better named `:query`